cpp/src/utilities/error_utils.hpp

```diff
@@ -25,6 +25,19 @@
 #define GDF_REQUIRE(F, S) \
   if (!(F)) return (S);
 
+/**---------------------------------------------------------------------------*
+ * @brief a version of GDF_REQUIRE for expressions of type `gdf_error` rather
+ * than booleans
+ *
+ * This macro is sort-of DEPRECATED.
+ *
+ *---------------------------------------------------------------------------**/
+#define GDF_TRY(...) do { \
```

Comment:
Can you give me an example of why this need to be variadic? I would think it should be the same as `RMM_TRY` above and just accept a single parameter.

Reply:
Actually, not really. Will make it a unary macro.

Reply:
Oh, and still need to add the other macro.